### PR TITLE
Issue #239 (Part 2/4) - Add Windows platform support

### PR DESCRIPTION
**Pull request by hs-apotell**
Issue #239 (Part 2)
Add Windows platform support
    
Highlights -
Add colon (:) as a legal character when parsing filepaths.
Also, fixed a few indenting issues.

### CODE REVIEW

**Comment by hs-apotell** [comment]
I don't see why the Presubmit tests are failing. The only edit in the file is to add that colon character in the regular expression. Every other change is just decorative.

Does 'colon' have a special meaning in regular expressions? It did work locally both on Unix and Windows. Will need some assistance with tclsh.

File: src/SourceCompile/generate_parser_listener.tcl [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 # Optional parameter to provide an output directory
 set DEST_DIR "."
-if [regexp {out_dir=([a-zA-Z0-9_/\.-]+)} $argv tmp DEST_DIR] {}
+if [regexp {out_dir=([a-zA-Z0-9_/\.-:]+)} $argv tmp DEST_DIR] {}
```
Comment (alaindargelas) [inline src/SourceCompile/generate_parser_listener.tcl:20]: Swap - and : 

if [regexp {out_dir=([a-zA-Z0-9_/\.:-]+)} $argv tmp DEST_DIR] {}

Comment (hs-apotell) [reply src/SourceCompile/generate_parser_listener.tcl]: Any reasoning as to why local builds on both Unix and Windows succeeded but ones on the test machines failed?

**Comment by alaindargelas** [comment]
I think you have not taken care of all the targets libraries dependance for all the executables, including the post release executwblectest.When running locally, remove the build dir completely and run "make release -j"You should replicate then locally the issues we see in the remote systems 

Sent from Yahoo Mail on Android 
 
  On Thu, May 21, 2020 at 13:13, HS<notifications@github.com> wrote:   


@hs-apotell commented on this pull request.

In src/SourceCompile/generate_parser_listener.tcl:
> @@ -17,7 +17,7 @@
 
 # Optional parameter to provide an output directory
 set DEST_DIR "."
-if [regexp {out_dir=([a-zA-Z0-9_/\.-]+)} $argv tmp DEST_DIR] {}
+if [regexp {out_dir=([a-zA-Z0-9_/\.-:]+)} $argv tmp DEST_DIR] {}

Any reasoning as to why local builds on both Unix and Windows succeeded but ones on the test machines failed?

File: src/SourceCompile/generate_parser_listener.tcl (hunk 1 above)
Comment (hs-apotell) [reply src/SourceCompile/generate_parser_listener.tcl]: Initially the build was failing and after my made my change it started working on Windows. So, I am certain I did build everything, not just once but multiple times. The version differences might be causing the error. On my machine its 8.6.8 and on the builder it's 8.6.0+9.

**Comment by alaindargelas** [comment]
From the .travis.yaml:
this is the proper way to test locally changes that impact the build process:
  - make release -j  - sudo make install  - make test_install  - make test-parallel
 I'll add a section in the README contributions to explain the process, that is missing.



    On Thursday, May 21, 2020, 01:37:42 PM PDT, alain.dargelas@gmail.com <alain.dargelas@gmail.com> wrote:  
 
 I think you have not taken care of all the targets libraries dependance for all the executables, including the post release executwblectest.When running locally, remove the build dir completely and run "make release -j"You should replicate then locally the issues we see in the remote systems 

Sent from Yahoo Mail on Android 
 
  On Thu, May 21, 2020 at 13:13, HS<notifications@github.com> wrote:   


@hs-apotell commented on this pull request.

In src/SourceCompile/generate_parser_listener.tcl:
> @@ -17,7 +17,7 @@
 
 # Optional parameter to provide an output directory
 set DEST_DIR "."
-if [regexp {out_dir=([a-zA-Z0-9_/\.-]+)} $argv tmp DEST_DIR] {}
+if [regexp {out_dir=([a-zA-Z0-9_/\.-:]+)} $argv tmp DEST_DIR] {}

Any reasoning as to why local builds on both Unix and Windows succeeded but ones on the test machines failed?

**Pull request merged**